jubilant/_juju.py

```diff
@@ -791,10 +839,25 @@ def refresh(
         if storage is not None:
             for k, v in storage.items():
                 args.extend(['--storage', f'{k}={v}'])
-        if trust:
+        if trust and not self._is_juju_2:
             args.append('--trust')
 
-        self.cli(*args)
+        mgr = (
+            contextlib.nullcontext()
+            if (not self._is_juju_2 or config is None)
+            else tempfile.TemporaryFile('w')  # noqa: SIM115
```

Comment:
Should this be `NamedTemporaryFile`? I don't think `TemporaryFile` is guaranteed to have a name. Also, I think it should probably have a `dir=self._temp_dir` argument like the other calls (due to the snap being unable to access the `/tmp` directory).

Reply:
I believe `TemporaryFile` is guaranteed to have a name, but that name might not be visible to the file system. But that is indeed an issue here, because the separate `juju` process needs to be able to see the file.